torch/optim/lr_scheduler.py

```diff
@@ -63,13 +65,23 @@ def get_lr(self):
     def step(self, epoch=None):
         # Raise a warning if old pattern is detected
         # https://github.com/pytorch/pytorch/issues/20124
-        if self.optimizer.step.called < 1:
-            warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "
-                          "In PyTorch 1.1.0 and later, you should call them in the opposite order: "
-                          "`optimizer.step()` before `lr_scheduler.step()`.  Failure to do this "
-                          "will result in PyTorch skipping the first value of the learning rate schedule."
-                          "See more details at "
-                          "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
+        if self._step_count == 1:
```

Comment:
Hello author @vfdev-5, sorry to trouble you, but just curious what's the reasoning behind checking whether it's equal to 1 here? Shouldn't it be `self._step_count == 0` (checking whether this is the first call to the function or not)? Thanks!

Reply:
Hello, the first call (`self._step_count == 0`) happens in `__init__` (line 43). See the details in the linked issue: https://github.com/pytorch/pytorch/issues/20124

Reply:
Ah I see, sorry didn't notice that... Thanks for clarifying!